src/gtpV2Codec/gtpV2Stack.cpp

```diff
@@ -394,6 +396,50 @@ GtpV2Stack::encodeMessage (GtpV2MessageHeader & msgHeader,
             }
             break;
         }
+        case EchoRequestMsgType:
+        {
+            if (data_p != NULL)
+            {
+                rc =
+               dynamic_cast<
+               EchoRequestMsg & >(msg).
+               encodeEchoRequestMsg(buffer,
+    			     *((EchoRequestMsgData *)
+        			     data_p));
+            }
+            else
+            { 
+                // Application has filled the data structure provided by the stack
+                rc = 
+                dynamic_cast<
+                EchoRequestMsg & >(msg).
+                encodeEchoRequestMsg (buffer,
+                            echoRequestStackData);
+            }
+            break;
+        }
+        case EchoResponseMsgType:
+        {
+            if (data_p != NULL)
+            {
+                rc =
+               dynamic_cast<
+               EchoResponseMsg & >(msg).
+               encodeEchoResponseMsg(buffer,
+    			     *((EchoResponseMsgData *)
+        			     data_p));
+            }
+            else
+            { 
+                // Application has filled the data structure provided by the stack
+                rc = 
+                dynamic_cast<
+                EchoResponseMsg & >(msg).
+                encodeEchoResponseMsg (buffer,
+                            echoResponseStackData);
+            }
+            break;
+        }
     }
 
     Uint16 endIndex = buffer.getCurrentIndex ();
```

Comment:
-    Uint16 messageLength = (endIndex - startIndex)+8;
+    Uint16 messageLength = (endIndex - startIndex)+4; // sequence number always present
+    if(msgHeader.teidPresent) {
+        messageLength += 4; 
+    }

This is what we should have to calculate messageLength

Reply:
Corrected in https://github.com/omec-project/Nucleus/pull/88/commits/006fce2d116447791ab76947fa75ebbc19aca9c9